internal/services/cragcast/cragcast.go

```diff
@@ -0,0 +1,227 @@
+	}
+
+	for _, hour := range hourlyForecasts {
+		cragcastDayIndex := len(cragcast.Days) - 1
+		// do we need a timezone conversion here?
+		day := hour.StartOfHour.Day()
+		if len(cragcast.Days) == 0 || cragcast.Days[cragcastDayIndex].Date.Day() != day {
+			// we are on a new day so we append a day to our cragcast
+			initialLowFahrenheit := float32(1000)
+			cragcast.Days = append(cragcast.Days, &DailyForecast{
+				Date:               hour.StartOfHour,
+				LowOfDayFahrenheit: &initialLowFahrenheit,
+				Hours:              []*HourlyForecast{},
+			})
+			cragcastDayIndex++
+		}
+
+		cragcastDay := cragcast.Days[cragcastDayIndex]
+		if hour.TotalPrecipMm != nil {
+			newPrecipMm := getPointerValueWithDefault(cragcastDay.TotalPrecipMm) + *hour.TotalPrecipMm
+			cragcastDay.TotalPrecipMm = &newPrecipMm
+		}
+
+		if hour.ChanceOfPrecip != nil && getPointerValueWithDefault(cragcastDay.ChanceOfPrecip) < *hour.ChanceOfPrecip {
+			cragcastDay.ChanceOfPrecip = hour.ChanceOfPrecip
+		}
+		if hour.TemperatureFahrenheit != nil && getPointerValueWithDefault(cragcastDay.LowOfDayFahrenheit) > *hour.TemperatureFahrenheit {
+			cragcastDay.LowOfDayFahrenheit = hour.TemperatureFahrenheit
+		}
+		if hour.TemperatureFahrenheit != nil && getPointerValueWithDefault(cragcastDay.HighOfDayFahrenheit) < *hour.TemperatureFahrenheit {
+			cragcastDay.HighOfDayFahrenheit = hour.TemperatureFahrenheit
+		}
+
+		if hour.Humidity != nil && getPointerValueWithDefault(cragcastDay.HumidityHigh) < *hour.Humidity {
+			cragcastDay.HumidityHigh = hour.Humidity
+		}
+
+		if hour.snowfallMm != nil {
+			newSnowfall := getPointerValueWithDefault(cragcastDay.snowfallMm) + *hour.snowfallMm
+			cragcastDay.snowfallMm = &newSnowfall
+		}
+
+		cragcastDay.Hours = append(cragcastDay.Hours, hour)
+	}
+
+	return cragcast, nil
+}
+
+func getProcessedHours(weatherClientForecast *noaaTypes.GridpointGeoJson) ([]*HourlyForecast, error) {
+	hours := []*HourlyForecast{}
+	// make a dictionary where times are the keys and values are hourly forecasts, loop through all sky cover, temp, precip etc...
+	// and add this info to each hour
+	// once that is done, loop through keys to make the hourly forecast and then sort by date
+
+	hoursMap := make(map[string]*HourlyForecast)
+
+	quantValuesToProcess := []QuantitativeValueType{TEMPERATURE, HUMIDITY, PRECIPITATION_PROBABILITY, QUANTITATIVE_PRECIPITATION, SKY_COVER, SNOWFALL_AMOUNT, WIND_DIRECTION, WIND_SPEED}
+
+	for _, quantValue := range quantValuesToProcess {
+		err := processQuantitativeValues(quantValue, weatherClientForecast.Properties.AdditionalProperties, hoursMap)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	for _, hourlyForecast := range hoursMap {
+		hours = append(hours, hourlyForecast)
+	}
+
+	sort.Slice(hours, func(i, j int) bool {
+		return hours[i].StartOfHour.Before(hours[j].StartOfHour)
+	})
+
+	return hours, nil
+}
+
+type QuantitativeValueType string
+
+const (
+	TEMPERATURE                QuantitativeValueType = "temperature"
```

Comment:
switch from all caps to (camelCase) and from `_` separate to [MixedCaps](https://go.dev/doc/effective_go#mixed-caps).

While it's technical ok to use all caps, it's pretty uncommon in Go -- see [examples in the standard library as an example](https://pkg.go.dev/os@go1.20.3#pkg-constants).

_Note: The `const`s that are all caps in the above example are to conform to POSIX naming_